dll/win32/newdev/wizard.c

```diff
@@ -579,6 +579,32 @@ AutoDriver(HWND Dlg, BOOL Enabled)
     IncludePath(Dlg, Enabled & IsDlgButtonChecked(Dlg, IDC_CHECK_PATH));
 }
 
+static INT CALLBACK
+BrowseCallbackProc(HWND hwnd, UINT uMsg, LPARAM lParam, LPARAM lpData)
+{
+    switch (uMsg)
+    {
+        case BFFM_INITIALIZED:
+        {
+            PCWSTR pszPath = ((PDEVINSTDATA)lpData)->CustomSearchPath;
+            SendMessageW(hwnd, BFFM_SETSELECTION, TRUE, (LPARAM)pszPath);
+            SendMessageW(hwnd, BFFM_ENABLEOK, 0, CheckBestDriver((PDEVINSTDATA)lpData, pszPath));
+            break;
+        }
+
+        case BFFM_SELCHANGED:
+        {
+            WCHAR szDir[MAX_PATH];
+            if (SHGetPathFromIDListW((LPITEMIDLIST)lParam, szDir))
+            {
+                return CheckBestDriver((PDEVINSTDATA)lpData, szDir);
+            }
```

Comment:
According to MSDN documentation, we should send `BFFM_ENABLEOK` message ourselves here. The code added in shell32 is a hack.

Reply:
This implementation has the same behavior as windows.

In BFFM_SELCHANGED message we check for best drive in current path
The button is enabled/dissable in BrsFolder_CheckValidSelection in wine code.
T

Reply:
`BrsFolder_CheckValidSelection` function does not allow to pass custom validation flags: https://source.winehq.org/git/wine.git/blob/5d1820c40bd93e6810f072a39745ee739ba99b81:/dlls/shell32/brsfolder.c#l469

If you read documentation at MSDN: https://msdn.microsoft.com/en-us/previous-versions/windows/desktop/legacy/bb762598(v=vs.85)

-- the `BFFM_SELCHANGED` message does not have a way to pass additional validation flags.

I have replaced it with a correct implementation.

Reply:
There is also an answer on SO about it: https://stackoverflow.com/a/31546444/4381493